src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -228,11 +237,31 @@ def set_timesteps(
                 )
 
             self.num_inference_steps = num_inference_steps
-
-            step_ratio = self.config.num_train_timesteps // self.num_inference_steps
-            timesteps = (np.arange(0, num_inference_steps) * step_ratio).round()[::-1].copy().astype(np.int64)
             self.custom_timesteps = False
 
+            # "linspace", "leading", "trailing" corresponds to annotation of Table 2. of https://arxiv.org/abs/2305.08891
+            if self.config.timestep_spacing == "linspace":
+                timesteps = np.linspace(0, self.config.num_train_timesteps - 1, num_inference_steps).round()[
+                    ::-1
+                ].copy().astype(np.int64)
```

Comment:
Need to be careful here as some schedulers use `float` as others use ints.

Reply:
How about performing the type-casting after all the conditions have been traversed?

Reply:
Good idea, I'll check it out.